File: 226-invert-binary-tree/invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if(root == null) return root;

        Queue<TreeNode> queue = new LinkedList();

        queue.add(root);

        while(!queue.isEmpty()){
            int levelSize = queue.size();

            for(int i = 0; i < levelSize;i++){
                TreeNode curr = queue.remove();

                TreeNode hold = curr.left;
                curr.left = curr.right;
                curr.right = hold;

                if(curr.left != null){
                    queue.add(curr.left);
                }

                if(curr.right != null){
                    queue.add(curr.right);
                }
            }

        }

        return root;

        
    }
}

File: 226-invert-binary-tree/invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root: return root

        queue = deque()

        queue.append(root)

        while queue:
            level_node = queue.popleft()

            hold = level_node.left
            level_node.left = level_node.right
            level_node.right = hold

            if level_node.left:
                queue.append(level_node.left)
            
            if level_node.right:
                queue.append(level_node.right)
        

        return root
        